doc/prompt_task_map.json

```diff
@@ -0,0 +1,130 @@
+{
+    "commonsense_qa": [
+        "task073_commonsenseqa_answer_generation"
+    ],
+    "dream": [
+        "task247_dream_answer_generation"
+    ],
+    "quail": [
+		"task887_quail_answer_generation"
+    ],
+    "quartz": [
+    ],
+    "social_i_qa": [
+        "task384_socialiqa_question_classification",
+		"task580_socialiqa_answer_generation"
```

Comment:
Hmm ... isn't `social_i_qa` a question answering task?

Reply:
if so, I am confused about why it is mapped to an "answer_generation" task.

Reply:
Yes, they have created the following tasks types:
- answer verification (task384_socialiqa_question_classification)
- multiple-choice question answering (task580_socialiqa_answer_generation)
- contextual question answering without options (missing task)
- question generation from the given context and answer (missing task)

Reply:
Could you share the pointers to these? 
> answer verification (task384_socialiqa_question_classification)
> multiple-choice question answering (task580_socialiqa_answer_generation)

Reply:
answer verification (task384_socialiqa_question_classification) -> [data in sheet](https://docs.google.com/spreadsheets/d/1wo637vMdwa-gEKT9T1eotOXSZiIW1rWyRJzyZcA73sc/edit#gid=1761964683&range=A994) [task](https://github.com/allenai/natural-instructions/blob/master/tasks/task384_socialiqa_question_classification.json)
multiple-choice question answering (task580_socialiqa_answer_generation) -> [data in sheet](https://docs.google.com/spreadsheets/d/1wo637vMdwa-gEKT9T1eotOXSZiIW1rWyRJzyZcA73sc/edit#gid=1761964683&range=A1189) [task](https://github.com/allenai/natural-instructions/blob/master/tasks/task580_socialiqa_answer_generation.json)

Reply:
Not our task; I am asking for the pointers to the prompt-source equivalent of these tasks.

Reply:
You can see the prompts in [Hosted version of PromptSource](https://bigscience.huggingface.co/promptsource).
or [here](https://github.com/bigscience-workshop/promptsource/tree/main/promptsource/templates/social_i_qa)

Reply:
"Check if a random answer is valid or not" prompt is answer verification and "Show choices and generate answer" prompt is multiple-choice question answering.

Reply:
Follow-up: so, does your script produce several task json files for [this](https://github.com/bigscience-workshop/promptsource/tree/main/promptsource/templates/social_i_qa) prompt file?

Reply:
Yes, it produces 6 task json files (One task for each prompt in the template).

Reply:
Okay. How can we (i.e., @yizhongw ) know which of these tasks (json file) correspond to `socialiqa_question_classification` or `socialiqa_question_classification`?

Reply:
I've added the prompt name and id as a key in the json files. 
Also, I can change this file to a prompt to task map.

Reply:
> I've added the prompt name and id as a key in the json files.

Maybe I'm missing something here. Does the name of these json files tell us whether they correspond to `socialiqa_question_classification` or  `socialiqa_question_classification`?


My understanding is that, here we have a 1-to-2 mapping (as opposed to 1-to-1 mapping). It's possible that I am missing something, in which case, help me see it! :)

Reply:
No, you're right! I think I can use better architecture here. 
Script gets the name of the dataset and generates a task for each prompt, and the name of the json files is based on the dataset and prompt names (e.g., task_socialiqa_Generate_answer).
The mapping is based on the datasets and our tasks and it hasn't mapped the tasks to the prompts. So, I need to add the prompt-task correspondence.

Reply:
Yeah exactly.